### PR TITLE
Add 24-hour clock setting

### PR DESCRIPTION
**Pull request by stefansundin**
I am sure there are plenty of things that might be wrong here, so please let me know what changes you would like to see.

## PR Checklist

- [x] I have checked that this PR is not a duplicate of an existing PR (open, closed or merged)
- [x] I have checked that this PR does not introduce a breaking change
- [ ] This PR introduces breaking changes and I have provided a detailed explanation below

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting)
- [ ] Refactoring (no functional changes)
- [ ] Documentation changes
- [ ] Other - Please describe:

## Fixes

Issue Number: I think this probably fixes https://github.com/lbryio/lbry-desktop/issues/5522.

## What is the current behavior?

Always shows 12-hour clock with AM/PM.

## What is the new behavior?

If the new `24-hour clock` checkbox in the settings is checked, the time is displayed in 24-hour format.

setting

video

transactions

Please let me know if there's another place that displays a date string with hours present.

## Other information

I don't think there's a foolproof way to detect 24-hour format preference from the browser/OS. https://stackoverflow.com/questions/27647918/detect-with-javascript-if-users-machine-is-using-12-hour-clock-am-pm-or-24-cl

Maybe there's a good way to do it in the Electron app, but I don't have much experience with that.

### CODE REVIEW

File: ui/redux/reducers/settings.js [hunk 1]
```diff
@@ -46,6 +46,7 @@ const defaultState = {
     [SETTINGS.HIDE_BALANCE]: false,
     [SETTINGS.OS_NOTIFICATIONS_ENABLED]: true,
     [SETTINGS.AUTOMATIC_DARK_MODE_ENABLED]: false,
+    [SETTINGS.CLOCK_24H]: false,
```
Comment (stefansundin) [inline ui/redux/reducers/settings.js:49]: I have no idea what this does, but the other settings were here so I added this one too.

**Comment by tzarebczan** [comment]
Thanks for the PR, we'll get a review on it soon. Can you also squash the commits?

**Comment by stefansundin** [comment]
> Thanks for the PR, we'll get a review on it soon. Can you also squash the commits?

I could do that, but first do you know about the "squash and merge" option? I think you'll find that it is better to squash using that, since 1. you get to edit the commit message, and 2. it preserves the commits in the PR which makes it easier to come back here and look at the changes separately at a later time. At least I think this is superior to manually squashing.

Please let me know if you still want me to squash, and please review it first so I know if there's something I should remove.

**Comment by seanyesmunt** [comment]
We tend to do "Rebase and merge". I think it's fine for you to not squash these since they are simple enough and there are only a few.

File: ui/redux/reducers/settings.js (hunk 1 above)
Comment (seanyesmunt) [reply ui/redux/reducers/settings.js:49]: This is the default value for this setting, which looks good

**Review by seanyesmunt — APPROVED** [review]
This is great! Thanks for the PR

**Comment by stefansundin** [comment]
Thanks for merging the PR. Looking forward to seeing it in production. :)

> We tend to do "Rebase and merge".

I think that technically, a single commit merged with "Rebase and merge" will have the exact same end result as many commits merged with "Squash and merge". In both cases there will not be a merge commit. Unless I'm missing something?

**Pull request merged**